app/services/content_block/find_and_replace_embed_codes_service.rb

```diff
@@ -29,16 +29,35 @@ def embed_content_references
       @embed_content_references ||= ContentBlockTools::ContentBlockReference.find_all_in_document(html)
     end
 
-    def identifiers
-      embed_content_references.map(&:identifier)
+    def unique_identifiers
+      embed_content_references.uniq.map(&:identifier)
     end
 
     def content_blocks
-      @content_blocks ||= begin
-        scope = ContentBlockManager::ContentBlock::Edition.current_versions
-        scope.where(document: { content_id: identifiers })
-             .or(scope.where(document: { content_id_alias: identifiers }))
+      @content_blocks ||= get_content_items_from_publishing_api.map do |result|
+        PublishingApiContentItem.new(result)
       end
     end
+
+    def get_content_items_from_publishing_api
+      Services
+        .publishing_api
+        .get_content_items(
+          content_id_aliases: unique_identifiers,
+          fields: %w[title content_id_aliases details document_type],
+          states: %w[published],
+        )
+        .fetch("results")
+    end
+
+    def render_block(content_block)
+      ContentBlockTools::ContentBlock.new(
+        document_type: content_block.fetch("document_type"),
+        content_id: content_block.fetch("content_id"),
+        title: content_block.fetch("title"),
+        details: content_block.fetch("details"),
+        embed_code: content_block.fetch("content_id_aliases").first.fetch("name"),
```

Comment:
`embed_code` here should come from the reference, so we know what exactly to render. For example, if the embed code is:

```
{{embed:content_block_contact:my-contact/addresses/bar}}
```

Content Block Tools knows to only render the `address` `bar`.